kubeadm/scripts/Install-Containerd.ps1

```diff
@@ -0,0 +1,158 @@
+<#
+.SYNOPSIS
+Installs ContainerD on a Windows machines in preperation for joining the node to a Kubernetes cluster.
+
+.DESCRIPTION
+This script
+- Verifies that Windows Features requried for running contianers are enabled (and enables then if they are not)
+- Downloads ContainerD binaries from from at the version specified.
+- Downloads Windows SND CNI plugins.
+- Sets up a basic nat networking config for ContainerD to use until another CNI is configured
+- Registers ContainerD as a windows service.
+
+.PARAMETER ContainerDVersion
+ContainerD version to download and use.
+
+.PARAMETER netAdapterName
+Name of network adapter to use when configuring basic nat network.
+
+.EXAMPLE
+PS> .\Install-Conatinerd.ps1
+
+#>
+
+Param(
+    [parameter(HelpMessage = "ContainerD version to use")]
+    [string] $ContainerDVersion = "1.4.1",
+    [parameter(HelpMessage = "Name of network adapter to use when configuring basic nat network")]
+    [string] $netAdapterName = "Ethernet"
+)
+
+$ErrorActionPreference = 'Stop'
+
+function DownloadFile($destination, $source) {
+    Write-Host("Downloading $source to $destination")
+    curl.exe --silent --fail -Lo $destination $source
+
+    if (!$?) {
+        Write-Error "Download $source failed"
+        exit 1
+    }
+}
+
+function CalculateSubNet {
```

Comment:
This is a complex calculation.  Do we have an example of what it does in docs here?  Maybe add a test?

Reply:
Let me add an example.
I can follow up with a test but since we don't have any other pester/powershell tests currently I think we can just add all that later.